mne_connectivity/spectral/time.py

```diff
@@ -458,137 +474,107 @@ def _spectral_connectivity(data, method, kernel, foi_idx,
-    psd = _smooth_spectra(psd, kernel)
-
-    def pairwise_coh(w_x, w_y):
-        s_xy = _compute_csd(w[:, w_y], w[:, w_x], weights)
-        s_xy = _smooth_spectra(s_xy, kernel)
-        s_xx = psd[:, w_x]
-        s_yy = psd[:, w_y]
-        out = np.abs(s_xy.mean(axis=-1, keepdims=True)) / \
-            np.sqrt(s_xx.mean(axis=-1, keepdims=True) *
-                    s_yy.mean(axis=-1, keepdims=True))
-        # mean inside frequency sliding window (if needed)
-        if isinstance(foi_idx, np.ndarray) and faverage:
-            return _foi_average(out, foi_idx)
+    if 'coh' in method:
+        # psd
+        if weights is not None:
+            psd = weights * w
+            psd = psd * np.conj(psd)
+            psd = psd.real.sum(axis=1)
+            psd = psd * 2 / (weights * weights.conj()).real.sum(axis=0)
         else:
-            return out
-
-    # define the function to compute in parallel
-    parallel, p_fun, n_jobs = parallel_func(
-        pairwise_coh, n_jobs=n_jobs, verbose=verbose, total=total)
+            psd = w.real ** 2 + w.imag ** 2
+            psd = np.squeeze(psd, axis=1)
 
-    return parallel(p_fun(s, t) for s, t in zip(source_idx, target_idx))
-
-
-def _plv(w, kernel, foi_idx, source_idx, target_idx, n_jobs, verbose, total,
-         faverage, weights):
-    """Pairwise phase-locking value.
+        # smooth
+        psd = _smooth_spectra(psd, kernel)
+    else:
+        psd = None
 
-    Input signal w is of shape (n_epochs, n_chans, n_tapers, n_freqs,
-    n_times)."""
-    def pairwise_plv(w_x, w_y):
-        s_xy = _compute_csd(w[:, w_y], w[:, w_x], weights)
-        exp_dphi = s_xy / np.abs(s_xy)
-        exp_dphi = _smooth_spectra(exp_dphi, kernel)
-        # mean over time
-        exp_dphi_mean = exp_dphi.mean(axis=-1, keepdims=True)
-        out = np.abs(exp_dphi_mean)
-        # mean inside frequency sliding window (if needed)
-        if isinstance(foi_idx, np.ndarray) and faverage:
-            return _foi_average(out, foi_idx)
-        else:
-            return out
+    # only show progress if verbosity level is DEBUG
+    if verbose != 'DEBUG' and verbose != 'debug' and verbose != 10:
+        total = None
 
     # define the function to compute in parallel
-    parallel, p_fun, n_jobs = parallel_func(
-        pairwise_plv, n_jobs=n_jobs, verbose=verbose, total=total)
+    parallel, my_pairwise_con, n_jobs = parallel_func(
+        _pairwise_con, n_jobs=n_jobs, verbose=verbose, total=total)
 
-    return parallel(p_fun(s, t) for s, t in zip(source_idx, target_idx))
+    return parallel(
+        my_pairwise_con(w, psd, s, t, method, kernel,
+                        foi_idx, faverage, weights)
+        for s, t in zip(source_idx, target_idx))
 
 
-def _pli(w, kernel, foi_idx, source_idx, target_idx, n_jobs, verbose, total,
-         faverage, weights):
-    """Pairwise phase-lag index.
+def _pairwise_con(w, psd, x, y, method, kernel, foi_idx,
+                  faverage, weights):
```

Comment:
Even tho this is a private function, do you mind adding a basic docstring mainly for internal documentation purposes?